components/hal/twai_hal_iram.c

```diff
@@ -63,7 +63,9 @@ static inline uint32_t twai_hal_decode_interrupt(twai_hal_context_t *hal_ctx)
     }
     //Transmit interrupt set whenever TX buffer becomes free
 #ifdef CONFIG_TWAI_ERRATA_FIX_TX_INTR_LOST
-    if ((interrupts & TWAI_LL_INTR_TI || hal_ctx->state_flags & TWAI_HAL_STATE_FLAG_TX_BUFF_OCCUPIED) && status & TWAI_LL_STATUS_TBS) {
+    if ((interrupts & TWAI_LL_INTR_TI || hal_ctx->state_flags & TWAI_HAL_STATE_FLAG_TX_BUFF_OCCUPIED)
+        && status & TWAI_LL_STATUS_TBS
+        && !(state_flags & TWAI_HAL_STATE_FLAG_BUS_OFF)) {
```

Comment:
In this case, we end up not generating the `TWAI_HAL_EVENT_TX_BUFF_FREE` event, which means the failed TX message that caused the Bus Off condition doesn't get handled in `twai_intr_handler_main()->twai_handle_tx_buffer_frame()`.